### PR TITLE
[DRAFT][runtime] Support RmsNorm operation

### PR DESCRIPTION
**Pull request by seockho-kim**
To support LLM on onert,

- Add RmsNorm Op to circle schema
- RmsNorm kernel implemented.

ONE-DCO-1.0-Signed-off-by: seockho.kim [seockho.kim@samsung.com](mailto:seockho.kim@samsung.com)

### CODE REVIEW

**Comment by seockho-kim** [comment]
For https://github.com/Samsung/ONE/issues/14089

File: runtime/onert/core/src/ir/operation/RmsNorm.cc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace ir
+{
+namespace operation
+{
+
+void RmsNorm::accept(OperationVisitor &v) const { v.visit(*this); }
+
+RmsNorm::RmsNorm(const OperandIndexSequence &inputs, const OperandIndexSequence &outputs,
+                 const Param &param)
+  : Operation{OperandConstraint::createExact(3u), inputs, outputs}, _param{param}
```
Comment (ragmani) [inline runtime/onert/core/src/ir/operation/RmsNorm.cc:31]: If a circle model without beta(optional input) can be created, you can use `OperandConstraint` in range?
`createExact(3u)` generates an error if we try to run a circle model without beta.
```suggestion
  : Operation{OperandConstraint::createInRange(2u, 3u), inputs, outputs}, _param{param}
```

[the remark this answers is not in the document]
Comment (ragmani) [reply runtime/onert/core/src/ir/operation/RmsNorm.cc:31]: As we talked offline, circle models with RMSNorm op always has beta input. So you don't need to modify it as I suggested.

File: runtime/onert/core/src/compiler/ShapeValidator.cc [hunk 2]
```diff
@@ -1123,5 +1123,22 @@ void ShapeValidator::visit(const ir::operation::LogSoftmax &node)
...
+    return;
+
+  const auto ifm_index{node.getInputs().at(ir::operation::RmsNorm::Input::INPUT)};
+  const auto gamma_index{node.getInputs().at(ir::operation::RmsNorm::Input::GAMMA)};
+  const auto beta_index{node.getInputs().at(ir::operation::RmsNorm::Input::BETA)};
+
+  OP_REQUIRES(operands.at(ifm_index).shape().rank() == 4);
+  OP_REQUIRES(operands.at(ifm_index).shape() == operands.at(ofm_index).shape());
+  OP_REQUIRES(operands.at(gamma_index).shape().rank() == 1);
+  OP_REQUIRES(operands.at(beta_index).shape().rank() == 1);
```
[the remark this answers is not in the document]
Comment (ragmani) [reply runtime/onert/core/src/compiler/ShapeValidator.cc:1140]: ditto

File: runtime/onert/core/src/ir/operation/RmsNorm.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (seockho-kim) [reply runtime/onert/core/src/ir/operation/RmsNorm.cc:31]: Thanks for the review. 
As we already talked, Fused RmsNorm operation has beta input 
(If it has no bias, FE sets default bias to 0.0)

**Comment by seockho-kim** [comment]
Done.

**Pull request closed without merging**